lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -40,8 +40,18 @@ def import_module(name):  # type: ignore[misc]
     reload_module = reload  # type: ignore[name-defined]  # pylint:disable=undefined-variable
 
 try:
-    from importlib.abc import TraversableResources
+    try:
+        # Available on Python >= 3.11
+        # We ignore the import error that will trigger when running mypy with
+        # older Python versions.
+        from importlib.resources.abc import TraversableResources  # type: ignore[import]
+    except ImportError:
+        # Used with Python 3.9 and 3.10 only
+        # This member is still available as an alias up until Python 3.14 but
+        # is deprecated as of Python 3.12.
+        from importlib.abc import TraversableResources
 except ImportError:
+    # Python < 3.9
```

Comment:
hrm, we should use the same thing here for `python_version='3.9', but I'm not sure if either mypy or our comment parser support stacking or "off-line" comments...

Reply:
The deprecated checker currently requires the marker at the start of a comment, and doesn't allow other comments on the same line, so we'd need to use this:

```suggestion
    # Python < 3.9
    # deprecated: description='TraversableResources fallback' python_version='3.8'
```

Reply:
I don't think we can do it for this codepath. This is for using the collection finder on py2 fwiw within ansible-test, so yes technically yes `Python < 3.9` but we need this fallback until we no longer support python2 for modules as well.

Reply:
This works, since ansible-test is aware that this code is effectively used on targets (due to unit tests and sanity tests).

The logic for the classification can be found here:

https://github.com/ansible/ansible/blob/56b67cccc52312366b9ceed02a6906452864e04d/test/lib/ansible_test/_internal/commands/sanity/__init__.py#L824

It's exposed to the deprecated pylint plugin here: 

https://github.com/ansible/ansible/blob/56b67cccc52312366b9ceed02a6906452864e04d/test/lib/ansible_test/_internal/commands/sanity/pylint.py#L262

Reply:
It's also worth pointing out that since this file is checked using the minimum target Python version, the other deprecation comment (added for Python 3.10 above) will be evaluated in that context as well. That means it won't report an error until Python 3.11 becomes the minimum on targets.